### PR TITLE
feat: filter test emails from prod emails in gov notify (FPLA-1837)

### PR DESCRIPTION
**Pull request by KiranVarma02**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/FPLA-1837

### Change description ###

filter test emails from prod emails in gov notify 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[* ] No
```

### CODE REVIEW

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/email/NotificationService.java [hunk 1]
```diff
@@ -13,15 +13,24 @@
...
-@RequiredArgsConstructor(onConstructor = @__(@Autowired))
 public class NotificationService {
     private final NotificationClient notificationClient;
     private final ObjectMapper mapper;
+    private final String environment;
+
+    @Autowired
+    public NotificationService(NotificationClient notificationClient,
+                               ObjectMapper mapper,
+                               @Value("${fpl.env}") String environment) {
```
Comment (tomaszpowroznik) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/service/email/NotificationService.java:24]: this is not correct, cannot be overridden with env variable configured in chart

Comment (KiranVarma02) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/service/email/NotificationService.java:24]: fixed charts file, thanks for help.

Comment (tomaszpowroznik) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/service/email/NotificationService.java:24]: if env variable is not present then app wont bootstrap, i believe it is fine, it is better to fail fast rather then produce wrong thing, @akomisarek , do you agree

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/PlacementSubmittedEventControllerTest.java [hunk 2]
```diff
@@ -119,13 +120,13 @@ void shouldNotifyHmctsAdminWhenAddingNewChildPlacementAndCtscIsDisabled() throws
                 NEW_PLACEMENT_APPLICATION_NOTIFICATION_TEMPLATE,
                 "admin@family-court.com",
                 expectedTemplateParameters(),
-                CASE_ID);
+                ENVIRONMENT + "/" + CASE_ID);
```
Comment (tomaszpowroznik) [inline service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/PlacementSubmittedEventControllerTest.java:123]: repeated 11 times in this class, can have separate variable

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/email/NotificationServiceTest.java [hunk 3]
```diff
@@ -17,15 +20,18 @@
 import static uk.gov.hmcts.reform.fpl.NotifyTemplates.PARTY_ADDED_TO_CASE_BY_EMAIL_NOTIFICATION_TEMPLATE;
 
 @ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = {NotificationService.class, NotificationClient.class, JacksonAutoConfiguration.class})
```
Comment (tomaszpowroznik) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/email/NotificationServiceTest.java:23]: NotificationClient is redundant here

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/email/NotificationServiceTest.java [hunk 4]
```diff
@@ -17,15 +20,18 @@
...
+@TestPropertySource(properties = {"fpl.env=localhost" })
 public class NotificationServiceTest {
     public static final String TEST_RECIPIENT_EMAIL = "test@example.com";
     public static final String REFERENCE = "12345L";
     public static final String TEMPLATE_ID = PARTY_ADDED_TO_CASE_BY_EMAIL_NOTIFICATION_TEMPLATE;
+    private static final String CASE_REFERENCE_WITH_ENVIRONMENT = "localhost/" + REFERENCE;
```
Comment (tomaszpowroznik) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/email/NotificationServiceTest.java:29]: could be


@TestPropertySource(properties = {"fpl.env=" + ENV})
public class NotificationServiceTest {
  ....
    static final String ENV = "TEST_ENV";
    private static final String CASE_REFERENCE_WITH_ENVIRONMENT = String.format("%s/%s", ENV, REFERENCE);

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/email/NotificationServiceTest.java [hunk 5]
```diff
@@ -17,15 +20,18 @@
 import static uk.gov.hmcts.reform.fpl.NotifyTemplates.PARTY_ADDED_TO_CASE_BY_EMAIL_NOTIFICATION_TEMPLATE;
 
 @ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = {NotificationService.class, NotificationClient.class, JacksonAutoConfiguration.class})
+@TestPropertySource(properties = {"fpl.env=localhost" })
 public class NotificationServiceTest {
     public static final String TEST_RECIPIENT_EMAIL = "test@example.com";
     public static final String REFERENCE = "12345L";
     public static final String TEMPLATE_ID = PARTY_ADDED_TO_CASE_BY_EMAIL_NOTIFICATION_TEMPLATE;
```
Comment (tomaszpowroznik) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/email/NotificationServiceTest.java:28]: can be private, class does not need public

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/CaseSubmissionControllerSubmittedTest.java [hunk 6]
```diff
@@ -75,8 +77,9 @@
     private static final String DISPLAY_AMOUNT_TO_PAY = "displayAmountToPay";
     private static final String SURVEY_LINK = "https://www.smartsurvey.co"
         + ".uk/s/preview/FamilyPublicLaw/44945E4F1F8CBEE3E10D79A4CED903";
+    private static final Long caseId = nextLong();
```
[the remark this answers is not in the document]
Comment (KiranVarma02) [reply service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/CaseSubmissionControllerSubmittedTest.java:80]: i just moved existing but updated it now.

File: service/src/main/resources/application.yaml [hunk 7]
```diff
@@ -223,6 +223,7 @@ fees-register:
 
 
 fpl:
+  env: 'localhost'
```
[the remark this answers is not in the document]
Comment (KiranVarma02) [reply service/src/main/resources/application.yaml:226]: done

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/email/NotificationServiceTest.java (hunk 3 above)
Comment (KiranVarma02) [reply service/src/test/java/uk/gov/hmcts/reform/fpl/service/email/NotificationServiceTest.java:23]: removed.

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/email/NotificationServiceTest.java (hunk 5 above)
Comment (KiranVarma02) [reply service/src/test/java/uk/gov/hmcts/reform/fpl/service/email/NotificationServiceTest.java:28]: done

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/email/NotificationServiceTest.java (hunk 4 above)
Comment (KiranVarma02) [reply service/src/test/java/uk/gov/hmcts/reform/fpl/service/email/NotificationServiceTest.java:29]: done

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ActionCaseManagementOrderControllerSubmittedTest.java [hunk 8]
```diff
@@ -74,6 +74,7 @@
     private static final UUID ID = randomUUID();
     private static final byte[] PDF = {1, 2, 3, 4, 5};
     private static final DocumentReference CMO_DOCUMENT = DocumentReference.buildFromDocument(document());
+    private static final String CASE_REFERENCE_WITH_ENVIRONMENT = "localhost/" + CASE_ID;
```
[the remark this answers is not in the document]
Comment (KiranVarma02) [reply service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ActionCaseManagementOrderControllerSubmittedTest.java:77]: I have to change it in 20 files, i will do it if you think its worth 😄

[the remark this answers is not in the document]
Comment (tomaszpowroznik) [reply service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ActionCaseManagementOrderControllerSubmittedTest.java:77]: that is just global rename, 20 sec of work:)

[the remark this answers is not in the document]
Comment (KiranVarma02) [reply service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ActionCaseManagementOrderControllerSubmittedTest.java:77]: haha true... duh.. i will do that now 👍

[the remark this answers is not in the document]
Comment (KiranVarma02) [reply service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ActionCaseManagementOrderControllerSubmittedTest.java]: done

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/email/NotificationService.java (hunk 1 above)
Comment (akomisarek) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/service/email/NotificationService.java:24]: Yes agreed :)

**Pull request merged**